From c7f78b972701445798394a46f48c4c7568d14ea0 Mon Sep 17 00:00:00 2001
From: Stefan Csomor <csomor@advancedconcepts.ch>
Date: Fri, 24 Oct 2025 17:13:50 +0200
Subject: [PATCH 11/48] bring mimetype.h includes to the same rule as
 mimetype.cpp does

also guard code that does not build on osx
---
 include/wx/gtk/mimetype.h | 12 +++++++-----
 src/gtk/mimetype.cpp      |  2 +-
 2 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/include/wx/gtk/mimetype.h b/include/wx/gtk/mimetype.h
index f3e0779e49..7037d1c557 100644
--- a/include/wx/gtk/mimetype.h
+++ b/include/wx/gtk/mimetype.h
@@ -12,10 +12,12 @@
 
 #include "wx/defs.h"
 
-#if defined(__UNIX__)
-#include "wx/unix/mimetype.h"
-#elif defined(__WINDOWS__)
-#include "wx/msw/mimetype.h"
+#if defined(__WINDOWS__)
+    #include "wx/msw/mimetype.h"
+#elif ( defined(__DARWIN__) )
+    #include "wx/osx/mimetype.h"
+#elif defined(__UNIX__)
+    #include "wx/unix/mimetype.h"
 #endif
 
 #if wxUSE_MIMETYPE
@@ -23,7 +25,7 @@
 class WXDLLIMPEXP_CORE wxGTKMimeTypesManagerImpl : public wxMimeTypesManagerImpl
 {
 protected:
-#if defined(__UNIX__)
+#if defined(__UNIX__) && !defined(__DARWIN__)
     wxString GetIconFromMimeType(const wxString& mime) override;
 #endif
 };
diff --git a/src/gtk/mimetype.cpp b/src/gtk/mimetype.cpp
index a7d681927b..7e2825488c 100644
--- a/src/gtk/mimetype.cpp
+++ b/src/gtk/mimetype.cpp
@@ -18,7 +18,7 @@
 #include "wx/gtk/private/string.h"
 #include "wx/gtk/private/object.h"
 
-#if defined(__UNIX__)
+#if defined(__UNIX__) && !defined(__DARWIN__)
 wxString wxGTKMimeTypesManagerImpl::GetIconFromMimeType(const wxString& mime)
 {
     wxString icon;
-- 
2.24.3 (Apple Git-128)

